pyannote/audio/applications/multilabel.py

```diff
@@ -0,0 +1,584 @@
+            default_module_name='pyannote.audio.labeling.tasks')
+
+        self.task_ = Task(protocol_name=protocol_name,
+                          preprocessors=self.preprocessors_,
+                          **self.config_['task'].get('params', {}))
+
+        # architecture
+        Architecture = get_class_by_name(
+            self.config_['architecture']['name'],
+            default_module_name='pyannote.audio.labeling.models')
+
+        params = self.config_['architecture'].get('params', {})
+        self.get_model_ = partial(Architecture, **params)
+
+        if hasattr(Architecture, 'get_frame_info'):
+            self.frame_info_ = Architecture.get_frame_info(**params)
+        else:
+            self.frame_info_ = None
+
+        if hasattr(Architecture, 'frame_crop'):
+            self.frame_crop_ = Architecture.frame_crop
+        else:
+            self.frame_crop_ = None
+
+    @classmethod
+    def from_train_dir(cls, protocol_name, train_dir, db_yml=None, training=False, use_der=False):
+        experiment_dir = dirname(dirname(train_dir))
+        app = cls(protocol_name, experiment_dir, db_yml=db_yml, training=training, use_der=use_der)
+        app.train_dir_ = train_dir
+
+        return app
+
+    @classmethod
+    def from_model_pt(cls, protocol_name, model_pt, db_yml=None, training=False, use_der=False):
+        train_dir = dirname(dirname(model_pt))
+        app = cls.from_train_dir(protocol_name, train_dir, db_yml=db_yml, training=training, use_der=use_der)
+        app.model_pt_ = model_pt
+        epoch = int(basename(app.model_pt_)[:-3])
+        app.model_ = app.load_model(epoch, train_dir=train_dir)
+        return app
+
+    def validate_init(self, protocol_name, subset='development'):
+        if self.label in self.task_.labels_spec["regular"]:
+            derivation_type = "regular"
+        elif self.label in self.task_.labels_spec["union"]:
+            derivation_type = "union"
+        elif self.label in self.task_.labels_spec["intersection"]:
+            derivation_type = "intersection"
+        else:
+            raise ValueError("%s not found in training labels : %s"
+                             % (self.label, self.task_.label_names))
+
+        protocol = get_protocol(protocol_name, progress=False,
+                                preprocessors=self.preprocessors_)
+        files = getattr(protocol, subset)()
+
+        self.pool_ = mp.Pool(mp.cpu_count())
+
+        # pre-compute features for each validation files
+        validation_data = []
+        for current_file in tqdm(files, desc='Feature extraction'):
+            # precompute features
+            if not isinstance(self.feature_extraction_, Precomputed): 
+                current_file['features'] = self.feature_extraction_(
+                    current_file)
+
+            if derivation_type == "regular":
+                current_file[self.label+"_ref"] = current_file["annotation"].subset([self.label])
```

Comment:
I would go with something like `current_file['multi_task'][label]` instead.
This will avoid any future conflict with preprocessors with the same name...

Reply:
Also, it is my understanding that there is only one `label` validated at a time.
Why do we need to add its name as prefix to `ref`?